### PR TITLE
xsos: init at 0.7.19

### PR DESCRIPTION
**Pull request by nixinator**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nixinator** [comment]
Xsos, nice little shell script, however i can't get the shell auto completion to work as mentioned in  
https://discourse.nixos.org/t/packaging-a-shell-script-that-has-completions/10290/3

so a little help appreciated.

File: pkgs/os-specific/linux/xsos/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    ethtool
+    pciutils
+    multipath-tools 
+    iproute
+  ];
+in 
+  
+stdenv.mkDerivation rec {
+  pname = "xsos";
+  version = "v0.7.19";
```
Comment (legendofmiracles) [inline pkgs/os-specific/linux/xsos/default.nix:25]:
```suggestion
  version = "0.7.19";
```

File: pkgs/os-specific/linux/xsos/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ryran";
+    repo = "xsos";
+    rev = version;
```
Comment (legendofmiracles) [inline pkgs/os-specific/linux/xsos/default.nix:30]:
```suggestion
    rev = "v${version}";
```

File: pkgs/os-specific/linux/xsos/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Summarize system info from sosreports";
+    homepage = "https://github.com/ryran/xsos";
+    license = licenses.gpl3;
+    maintainers = [ maintainers.nixinator ];
+  };
+}
+
```
Comment (legendofmiracles) [inline pkgs/os-specific/linux/xsos/default.nix:54]:
```suggestion
```

**Comment by legendofmiracles** [comment]
The bash completion isn't installed because you never call the postInstall phase from installPhase. I've fixed it for you in [this patch](http://0x0.st/-hoi.txt). I also added busybox as a dependency because otherwise xsos couldn't find `runlevel`

**Comment by nixinator** [comment]
> The bash completion isn't installed because you never call the postInstall phase from installPhase. I've fixed it for you in [this patch](http://0x0.st/-hoi.txt). I also added busybox as a dependency because otherwise xsos couldn't find `runlevel`

thanks for that. Hopefully busybox doesn't pollute the path with different tools. I've seen this happen with installing stuff via nix-env , you think your getting a tool from core utils etc etc, but it's actually coming from busy box itself due to name collisions on the path... ordering is important here, and can be darn confusing..... Is there a version of runlevel that isn't the cut down, limited functionality that busy-box provides.

**Comment by legendofmiracles** [comment]
The only other packages that provide `runlevel` are `sysvinit` are `molly-guard`

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 147771` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xsos</li>
  </ul>
</details>

**Review by legendofmiracles — APPROVED** [review]

**Comment by nixinator** [comment]
well `nixshell -p nixpkgs-review` is rather nice isn't it. This force is strong with mic92

**Pull request merged**